icu4c/source/test/intltest/unitstest.cpp

```diff
@@ -222,61 +301,60 @@ void unitsTestDataLineFn(void *context, char *fields[][2], int32_t fieldCount, U
                      quantity.length(), quantity.data(), x.length(), x.data(), y.length(), y.data(),
                      expected, commentConversionFormula.length(), commentConversionFormula.data());
 
-    // WIP(hugovdm): hook this up to actual tests.
-
-    // // Convertibility:
-    // MaybeStackVector<MeasureUnit> units;
-    // units.emplaceBack(sourceUnit);
-    // units.emplaceBack(targetUnit);
-    // const auto &conversionRateInfoList = getConversionRatesInfo(units, status);
-    // if (status.errIfFailureAndReset("getConversionRatesInfo(...)")) return;
+    // Convertibility:
+    MaybeStackVector<MeasureUnit> units;
+    units.emplaceBack(sourceUnit);
+    units.emplaceBack(targetUnit);
+    const auto &conversionRateInfoList = getConversionRatesInfo(units, status);
+    if (status.errIfFailureAndReset("getConversionRatesInfo(...)")) return;
 
-    // auto actualState = checkUnitsState(sourceUnit, targetUnit, conversionRateInfoList, status);
-    // if (status.errIfFailureAndReset("checkUnitsState(<%s>, <%s>, ...)", sourceUnit.getIdentifier(),
-    //                                 targetUnit.getIdentifier())) {
-    //     return;
-    // }
+    auto actualState = checkConvertibility(sourceUnit, targetUnit, conversionRateInfoList, status);
+    if (status.errIfFailureAndReset("checkConvertibility(<%s>, <%s>, ...)", sourceUnit.getIdentifier(),
+                                    targetUnit.getIdentifier())) {
+        return;
+    }
 
-    // CharString msg;
-    // msg.append("convertible: ", status)
-    //     .append(sourceUnit.getIdentifier(), status)
-    //     .append(" -> ", status)
-    //     .append(targetUnit.getIdentifier(), status);
-    // if (status.errIfFailureAndReset("msg construction")) return;
+    CharString msg;
+    msg.append("convertible: ", status)
+        .append(sourceUnit.getIdentifier(), status)
+        .append(" -> ", status)
+        .append(targetUnit.getIdentifier(), status);
+    if (status.errIfFailureAndReset("msg construction")) return;
 
-    // unitsTest->assertTrue(msg.data(), actualState != UNCONVERTIBLE);
+    unitsTest->assertTrue(msg.data(), actualState != UNCONVERTIBLE);
```

Comment:
Probably better to assertEquals.  You will need to make sure the int32_t overload of assertEquals gets called (instead of the UBool version), which you can do with the help of static_cast.